src/ORM/DataList.php

```diff
@@ -1305,15 +1320,16 @@ public function eagerLoad(...$relations): static
                 $message = "Eager loading only supports up to 3 levels of nesting, passed $count levels - $relation";
                 throw new InvalidArgumentException($message);
             }
-            for ($i = 0; $i < $count; $i++) {
-                if ($i === 0) {
-                    $arr[] = $parts[$i];
-                } else {
-                    $arr[] = $arr[count($arr) - 1] . '.' . $parts[$i];
-                }
+            // Add each relation in the chain as its own entry to be eagerloaded
+            // e.g. for "Players.Teams.Coaches" you'll have three entries:
+            // "Players", "Players.Teams", and "Players.Teams.Coaches
+            $usedParts = [];
+            foreach ($parts as $part) {
+                $usedParts[] = $part;
+                $arr[] = implode('.', $usedParts);
```

Comment:
It wasn't at all clear what was happening here - this change doesn't improve performance but it does improve code readability.